models/sharky/solana/sharky_solana_events.sql

```diff
@@ -80,24 +80,40 @@ WITH sharky_txs AS (
         LEFT JOIN sol_price p
             ON p.minute = date_trunc('minute', filtered_txs.block_time)
     ),
-    events AS (
+    decoded_events AS (
         SELECT *,
+            -- We need to decode the instruction data to identify the instruction since the payload changes the base58 encoded string
+            base58_decode(sharky_instructions[0].data) AS first_ix_data
```

Comment:
this way we decode the ix_data only once, instead of decoding it every time in the CASE...WHEN statement at check